Source/Bifrost.FluentValidation/MetaData/ValidationMetaDataGenerator.cs

```diff
@@ -72,19 +79,21 @@ void GenerateForValidator(IValidator inputValidator, TypeMetaData metaData, stri
 
             foreach (var member in members)
             {
-                var rules = descriptor.GetRulesForMember(member.Key);
-                foreach (var rule in rules)
+                var isModelRule = member.Key == ModelRule<string>.ModelRulePropertyName || member.Key == string.Empty;
+                var currentKey = (shouldUseParentKey || isModelRule) ? parentKey : GetKeyForMember(parentKey, member);
+                var isConcept = new Lazy<bool>(() => IsConcept(genericArguments, isModelRule, member));
+
+                foreach (var rule in descriptor.GetRulesForMember(member.Key))
                 {
                     foreach (var validator in rule.Validators)
                     {
-                        var isModelRule = member.Key == ModelRule<string>.ModelRulePropertyName;
-                        var currentKey = GetKeyForMember(parentKey, isParentConcept, isParentModelRule, member, isModelRule);
-
-                        if (validator is ChildValidatorAdaptor)
+                        var childValidatorAdaptor = validator as ChildValidatorAdaptor;
+                        if (childValidatorAdaptor != null)
                         {
-                            GenerateForChildValidator(metaData, genericArguments, member, validator, isModelRule, currentKey);
+                            var childValidator = childValidatorAdaptor.GetValidator(NullContext);
```

Comment:
Larger incompatibility in FluentValidation: childValidatorAdaptor.Validator was changed to childValidatorAdaptor(context). Just using a dummy context works fine.

Also cleaned up this file somewhat.